internal/txprocessor/txprocessor.go

```diff
@@ -81,6 +87,11 @@ func (r TXProcessor) submitTxWithProofs(
 	hash := hex.EncodeToString(tmtypes.Tx(block.Tx.Data).Hash())
 	neutronTxHash, err := r.submitter.SubmitTxProof(ctx, queryID, block)
 	if err != nil {
+		// check error with regexp
+		if !r.ignoreErrorsRegexp.MatchString(err.Error()) {
+			r.logger.Error("failed to submit tx proof", zap.Error(err))
+			os.Exit(3)
```

Comment:
the only thing is bothering me
1) We ignore all our graceful shutdown
2) I reuse ProcessAndsubmit https://github.com/neutron-org/neutron-query-relayer/blob/feat/resubmit-failed-txs/internal/webserver/webserver.go#L87 which in case of network/gas error just stops the daemon with os.Exit(3)

I would consider to pass a critical error up on the stack

Reply:
absolutely so, we shouldn't just os.Exit() here